unity-client/Assets/Scripts/MainScripts/DCL/Environment/Environment.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System.ComponentModel;
+using DCL.Controllers;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+namespace DCL
+{
+    public class Environment
+    {
+        public static Model i = new Model();
+        private static System.Func<MessagingContext> messagingBuilder;
+        private static System.Func<PlatformContext> platformBuilder;
+        private static System.Func<WorldRuntimeContext> worldRuntimeBuilder;
+
+        public static void SetupWithDefaults()
+        {
+            Setup(
+                messagingBuilder: MessagingContextFactory.CreateDefault,
+                platformBuilder: PlatformContextFactory.CreateDefault,
+                worldRuntimeBuilder: WorldRuntimeContextFactory.CreateDefault
+            );
+        }
+
+        public static void Setup(System.Func<MessagingContext> messagingBuilder,
+            System.Func<PlatformContext> platformBuilder,
+            System.Func<WorldRuntimeContext> worldRuntimeBuilder)
+        {
+            Environment.messagingBuilder = messagingBuilder;
+            Environment.platformBuilder = platformBuilder;
+            Environment.worldRuntimeBuilder = worldRuntimeBuilder;
+            Setup();
+        }
```

Comment:
We pass `Func` as builder methods in order to be able to reset the environment state with the same implementations.

Reply:
I love the idea of the builders. Really well thought!